docs/ref/models/querysets.txt

```diff
@@ -2399,6 +2399,47 @@ Class method that returns an instance of :class:`~django.db.models.Manager`
 with a copy of the ``QuerySet``’s methods. See
 :ref:`create-manager-with-queryset-methods` for more details.
 
+
+``explain()``
+~~~~~~~~~~~~~
+
+.. versionadded: 2.1
+
+.. method:: explain(verbose=False, format=None)
+
+Returns the execution plan of the current queryset as a string. This details
+how the database would execute the given query, highlighting any indexes or
+joins used. For example, when using ``postgresql``::
+
+    >>> print(Blog.objects.filter(title="My Blog").explain())
+    Seq Scan on blog  (cost=0.00..35.50 rows=10 width=12)
+      Filter: (title = 'My Blog'::bpchar)
+
+Some databases (like MySQL and Postgres) accept flags that can return more
```

Comment:
Postgres → PostgreSQL

Reply:
Should we list the available formats and options in a table and/or provide links to documentation provided by vendors?

Reply:
Hmm, I'm not sure, do we do this anywhere else? I could add it, but I'd rather people search for the documentation for the correct version/vendor they are using.

Reply:
Feature support has been done before for the stuff in `django.contrib.gis`, but perhaps it is a little too much for here. We could link to documentation, but again that is difficult as most vendors have horrible version-specific URLs.

Only PostgreSQL is nice: https://www.postgresql.org/docs/current/static/sql-explain.html :heart: